src/GNDStk/gnds-utility.hpp

```diff
@@ -0,0 +1,231 @@
+
+// -----------------------------------------------------------------------------
+// Macros, variables, enums
+// -----------------------------------------------------------------------------
+
+// printval
+// printvar
+// Are the same, so we don't need to remember the exact terminology
+#define printval(var) std::cout << #var ": [" << var << "]" << std::endl
+#define printvar(var) printval(var)
+
+// indent
+// Number of spaces of indentation you want, in the output of certain types
+inline int indent = 3;
```

Comment:
Please make this `constexpr`.

Reply:
This is intentionally non-const, and is intended to be a user-settable variable. Some people like 2-space indentation, others 3-space, others 4-space. (More than that, I've seldom seen.) Just set GNDStk::indent in one place in your code, and henceforth, anything written to XML, to JSON, or to my internal tree-dump format, will use *that* number of spaces. For things like this, I've found that a nice global setting is more convenient for people that, say, having an optional argument in various write() functions here and there. Not to mention that stream output - operator>> - doesn't allow for an additional arguments. Even optional ones, and even if the operator>> would be called in functional form rather than operator form.

Reply:
I mean operator<< here.